File: 2020/2020W05/preppindataw05.R
library(dplyr)
library(tidyr)
library(readr)

final <- read_csv("E:/PD 2020 Wk 5 Input.csv",
                  col_types = cols(HTf = col_double())) %>%
  drop_na(`HTf`) %>%
  mutate('Rank' = rank(desc(`Diff`), ties.method='min')) %>%
  group_by(`Venue`) %>% 
  summarise('Number of Games' = n(),
            'Best Rank (Standard Competition)' = min(`Rank`),
            'Worst Rank (Standard Competition)' = max(`Rank`),
            'Avg Rank (Standard Competition)' = mean(`Rank`)) %>%
  ungroup() 

View(final)

File: 2020/2020W05/PreppindataW05.py
import pandas as pd
import re

final = pd.read_csv(r'E:/PD 2020 Wk 5 Input.csv')
final['HTf'] = final['HTf'].apply(lambda x: None if re.search('(\D)',x) else x)
final = final.dropna(subset=['HTf'])[['Diff','Venue']]
final['Rank'] = final['Diff'].rank(ascending=False, method='min').astype(int)
final = final.groupby('Venue',as_index=False).agg({'Diff':[('Number of Games','count')],
                                                'Rank':[('Best Rank (Standard Competition)','min'),
                                                        ('Worst Rank (Standard Competition)','max'),
                                                        ('Avg Rank (Standard Competition)','mean')]})
final.columns = [col[1] if col[1] else col[0] for col in final.columns.values]
